gap/InternalExtensions.g

```diff
@@ -0,0 +1,190 @@
+# HELPER FUNCTIONS, EXTENSIONS 
+
+# Dependencies:
+# - Internal.g
+# - kbmag package 1.5.10+
+
+# Constructs rewriting systems (rws) via Knuth-Bendix completion alogorithm
+# and stores it in cache.
+rwsFunc@ := MemoizePosIntFunction(function(ID)
+	local G, ORkbrws, maxlen, kbrws, mhom, mon, catch, lenRules, lmax;
+	
+	# Get needed input though options:
+	# --------------------------------
+	lmax := ValueOption("lmax");
+	G := ValueOption("group");
+
+	# ---------------------
+	# Set up kbrws options:
+ 	# ---------------------
+
+	# maximal word length left and right
+	# (somewhat arbitrary due to the nature of the algorithm (see documentation)
+	#  however, this is a rather stable, minimal choice)
+	maxlen := [lmax, 2*(1+lmax)]; 
+       	
+	# reduce group to monoid
+	mhom := IsomorphismFpMonoid(G);
+       	mon := Image(mhom);
+      	kbrws := KBMAGRewritingSystem(mon);	
+	
+	# ---------------
+	# Adjust options:
+	# ---------------
+	# in order to halt in a reasonable amount of time
+
+	# SetOrderingOfKBMAGRewritingSystem(kbrws, "recursive"); # alternative method
+       	ORkbrws := OptionsRecordOfKBMAGRewritingSystem(kbrws);
+   	ORkbrws.maxstoredlen := maxlen; # set maximal word length left and right
+
+	# ---------------------------------------
+       	# Construct (probably non-confluent) rws:
+	# ---------------------------------------
+
+	# occuring errors are "catched" (assuming SimplifyWord@ uses BreakOnError := false when entered) 
+       	catch := CALL_WITH_CATCH(KnuthBendix, [kbrws])[1];; # alternative: use ErrorCount
+	
+	# ---------------------------------------------
+	# Throw warning due to possibly occured errors:
+	# ---------------------------------------------
+
+	if not catch then
+		lenRules := Length(Rules(kbrws));
+
+		# error due the above defined options occured
+		# (or unexpected error occured) TODO: catch unexpected error separately
+		if Length(Rules(kbrws)[lenRules, 1]) + Length(Rules(kbrws)[lenRules, 2]) > maxlen[1] + maxlen[2] then
+			Print("#WARNING: simplification was unsuccessful; non-simplified words will be used.\n");
+			Print(StringFormatted("Try setting simplify to a value above {}.", lmax + 1));
+		
+		# error due to missing configurations in the package kbmag occured 
+		# (or (highly unlikely) possibly due to exceeding 65535 number of generators)
+		else
+			Print("#WARNING: maximal number of genartors have been exceeded; non-simplified words will be used.\n");
+			Print("Please follow the instructions in chapter ?? in order to use simplify for groups with more than 127 generators.");
+		fi;
+		return catch;
+	else
+		return [kbrws, mhom, mon, G];
+	fi;
+end );
+
+# SimplifyWord@ function extensions which add to the original brute force method
+# a rewriting system method via the Knuth-Bendix completion algorithm 
+SimplifyWord@ := function(G, w, lmax)
```

Comment:
Does this simplify overwrite the other definition?

Reply:
Yes. Also note that the corresponding file will be loaded after Internal.g has been loaded.